File: app/src/main/java/com/icebem/akt/util/RandomUtil.java
package com.icebem.akt.util;

public class RandomUtil {
    public static final int RANDOM_P = 5;
    public static final int RANDOM_T = 150;

    private static int random(int i, int r) {
        return i + (int) (Math.random() * r * 2) - r;
    }

    public static int randomP(int p) {
        return random(p, RANDOM_P);
    }

    public static int randomT(int t) {
        return random(t, RANDOM_T);
    }
}

File: app/src/main/java/com/icebem/akt/util/RandomUtil.kt
package com.icebem.akt.util

object RandomUtil {
    const val DELTA_POINT = 5
    private const val DELTA_TIME = 150

    private fun random(i: Int, delta: Int): Int = if (i > delta) i + (Math.random() * delta * 2).toInt() - delta else i

    fun randomIndex(length: Int): Int = (Math.random() * length).toInt()

    fun randomPoint(point: Int): Int = random(point, DELTA_POINT)

    fun randomTime(time: Int): Long = random(time, if (time > DELTA_TIME) DELTA_TIME else DELTA_POINT).toLong()
}